src/test/java/net/snowflake/client/jdbc/BindingDataLatestIT.java

```diff
@@ -122,57 +136,189 @@ public void testTimestampBindingWithNTZType() throws SQLException {
+        try (PreparedStatement prepStatement =
+            connection.prepareStatement("insert into regularinsert values (?,?,?,?)")) {
+          for (int i = 1; i <= 6; i++) {
+            prepStatement.setInt(1, 1);
+            prepStatement.setTimestamp(2, currT);
+            prepStatement.setTimestamp(3, currT);
+            prepStatement.setTimestamp(4, currT);
+            prepStatement.addBatch();
+          }
+          prepStatement.executeBatch();
+        }
+
+        // insert using stage binding
+        try (PreparedStatement prepStatement =
+            connection.prepareStatement("insert into stageinsert values (?,?,?,?)")) {
+          statement.execute("ALTER SESSION SET CLIENT_STAGE_ARRAY_BINDING_THRESHOLD = 1");
+          prepStatement.setInt(1, 1);
+          prepStatement.setTimestamp(2, currT);
+          prepStatement.setTimestamp(3, currT);
+          prepStatement.setTimestamp(4, currT);
+          prepStatement.addBatch();
+          prepStatement.executeBatch();
+        }
+
+        // Compare the results
+        try (ResultSet rs1 = statement.executeQuery("select * from stageinsert");
+            ResultSet rs2 = statement.executeQuery("select * from regularinsert")) {
+          rs1.next();
+          rs2.next();
+
+          assertEquals(rs1.getInt(1), rs2.getInt(1));
+
+          // Check that all the values are the same.
+          assertEquals(rs1.getTimestamp(2), rs1.getTimestamp(3));
+          assertEquals(rs1.getTimestamp(3), rs1.getTimestamp(4));
+
+          assertEquals(rs1.getTimestamp(2), rs2.getTimestamp(2));
+          assertEquals(rs1.getTimestamp(3), rs2.getTimestamp(3));
+          assertEquals(rs1.getTimestamp(4), rs2.getTimestamp(4));
+        }
+      } finally {
+        statement.execute("drop table if exists stageinsert");
+        statement.execute("drop table if exists regularinsert");
+        TimeZone.setDefault(origTz);
+      }
+    }
+  }
+
+  /**
+   * Test that stage binding and regular binding insert and return the same value for timestamp_ltz when the local timezone has the daylight saving.
+   *
+   * When CLIENT_TIMESTAMP_TYPE_MAPPING setting is mismatched with target data
+   * type (.e.g MAPPING=LTZ and insert to NTZ or MAPPING=NTZ and insert to TZ/LTZ
+   * there could be different result as the timezone offset is applied on client
+   * side and removed on server side. This only occurs around the boundary of
+   * daylight-savings and the difference from the source data would be one hour.
+   * Both regular binding and stage binding have such issue but they also behave
+   * diffently, for some data only regular binding gets the extra hour while
+   * sometime only stage binding does.
+   * The workaround is to use CLIENT_TIMESTAMP_TYPE_MAPPING=LTZ to insert
+   * LTZ/TZ data and use CLIENT_TIMESTAMP_TYPE_MAPPING=NTZ to insert NTZ data.
+   *
+   * @throws SQLException
+   */
+  @Test
+  @ConditionalIgnoreRule.ConditionalIgnore(condition = RunningOnGithubAction.class)
```

Comment:
let's add a comment why we cannot run it on Github (or allow the tests to run on GH also)